### ISSUE TITLE
Recipe descriptions

### ISSUE THREAD

**Issue by norm** [devfort/nosebag#4]
Many recipes are presented with prose descriptions. I suggest a key or keys at the top of the recipe object to represent this. Either keys like `description` / `introduction` / `postscript` / `history`... or a single key that contains an object.

The most common across my recipe books is a paragraph or several before the ingredients and steps, but I have also seen both a personal introduction to a recipe, then a description of it, the ingredients, the steps, then more prose afterward such as cooking tips and alternatives.

I would suggest they contain free text for the renderer to interpret as it will (eg I will be adding markdown to mine).

**Comment by jaylett** [comment]
I'd be happy with adding description. The problem with adding more of them is that it gets into physical presentation, and the entire point was to avoid that as much as possible. ("History" is an interesting one, but "postscript" is not really structural.)

I'm 50/50 on whether a text blob should be treated as plain text and to use an object with typing for anything else. Improves compatibility, makes things more complex. The main problem being that if you're stuffing Markdown in a text blob, nosebag almost certainly won't care and it'll look a little weird. Although at least rendering Markdown as text doesn't look terrible; it's not like you're going to stuff docbook in there or something. Maybe the rule should be "will not look terrible if rendered as plain text with newlines preserved". Or would you start splitting lines within Markdown on sentence structure to make the diffs easier? In which case we could advocate simple unwrapping (similar to the Django `|linebreaks` filter).

(I'm thinking for instance of extending nosebag for tablet use while cooking at forts, going back to the ideas of omnom.)

**Comment by norm** [comment]
Yeah, I have been trying to find the more involved recipes to see what doesn’t fit. I’d be happy to pare it back to a singular description. In the absence of more structure, you can have that contain subheadings. I just have seen some that are broken down into separate areas and presented around the recipe itself, so I thought it wise to mention.

We have precedent in ingredient amounts for simple or complex keys as needed, so I’d be happy with plain text vs some encoded representation with a content-type.

I’d also be happy with the default being to treat it Markdown, given mostly it’s just about paragraphs.

**Comment by jaylett** [comment]
I wouldn't be completely happy with the default being to treat as Markdown, because that's a burden on nosebag (and subsequent clients) that someone will have to implement. And I don't know which of the 48000 npm modules is the "right" one to use.

**Comment by norm** [comment]
Draft in https://github.com/devfort/nosebag/pull/10
